katdal/datasources.py

```diff
@@ -249,35 +250,51 @@ def _shorten_key(telstate, key):
     return ''
 
 
-def _infer_chunk_store(url_parts, telstate, chunk_name=None,
+def _infer_chunk_store(url_parts, telstate, npy_store_path=None,
                        s3_endpoint_url=None, **kwargs):
-    """Construct chunk store automatically from dataset URL and telstate."""
-    data_path = store_path = ''
-    try:
-        if url_parts.scheme == 'file':
-            # Look for adjacent data directory (presumably containing NPY files)
-            rdb_path = os.path.abspath(url_parts.path)
-            store_path = os.path.dirname(os.path.dirname(rdb_path))
-            if not store_path:
-                store_path = os.path.curdir
-            try:
-                chunk_name = chunk_name or telstate['chunk_name']
-            except KeyError as e:
-                raise StoreUnavailable('[{}] {}'.format(type(e).__name__, e))
-            else:
-                data_path = os.path.join(store_path, chunk_name)
+    """Construct chunk store automatically from dataset URL and telstate.
+
+    Parameters
+    ----------
+    url_parts : :class:`urlparse.ParseResult` object
+        Parsed dataset URL
+    telstate : :class:`katsdptelstate.TelescopeState` object
+        Telescope state
+    npy_store_path : string, optional
+        Top-level directory of NpyFileChunkStore (overrides the default)
+    s3_endpoint_url : string, optional
+        Endpoint of S3 service, e.g. 'http://127.0.0.1:9000' (overrides default)
+    kwargs : dict, optional
+        Extra keyword arguments, typically meant for other methods and ignored
+
+    Returns
+    -------
+    store : :class:`katdal.ChunkStore` object
+        Chunk store for visibility data
+
+    Raises
+    ------
+    KeyError
+        If telstate lacks critical keys
+    :exc:`katdal.chunkstore.StoreUnavailable`
+        If the chunk store could not be constructed
+    """
+    # Use overrides if provided, regardless of URL and telstate (NPY first)
+    if npy_store_path and os.path.isdir(npy_store_path):
+        return NpyFileChunkStore(npy_store_path)
+    if s3_endpoint_url:
+        return S3ChunkStore.from_url(s3_endpoint_url, **kwargs)
+    # NPY chunk store is an option if the dataset is an RDB file
+    if url_parts.scheme == 'file':
+        # Look for adjacent data directory (presumably containing NPY files)
+        rdb_path = os.path.abspath(url_parts.path)
+        store_path = os.path.dirname(os.path.dirname(rdb_path))
+        if not store_path:
+            store_path = os.path.curdir
```

Comment:
Under what conditions could `store_path` be false, given that you started with an absolute path and `os.path.dirname('/') == '/'`?